src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/linker/JdbcConnectionLinker.java

```
@@ -0,0 +1,60 @@
+import com.ca.apim.gateway.cagatewayexport.tasks.explode.bundle.entity.StoredPasswordEntity;
+
+import javax.inject.Singleton;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.ca.apim.gateway.cagatewayexport.util.gateway.VariableUtils.extractVariableName;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Singleton
+public class JdbcConnectionLinker implements EntityLinker<JdbcConnectionEntity> {
+
+    private Pattern secpassPattern = Pattern.compile("secpass.(.+?).plaintext");
+
+    @Override
+    public void link(JdbcConnectionEntity entity, Bundle bundle, Bundle targetBundle) {
+        // check if the password is specified and is a reference to a stored password
+        if (isEmpty(entity.getPasswordRef())) {
+            return;
+        }
+
+        String storedPasswordReference = extractVariableName(entity.getPasswordRef());
+        if (isEmpty(storedPasswordReference)) {
+            return;
+        }
+
+        Matcher matcher = secpassPattern.matcher(storedPasswordReference);
+        if (matcher.matches()) {
+            // the middle string between secpass and plaintext will be the stored password name
+            String storedPasswordName = matcher.group(1);
+            final StoredPasswordEntity storedPassword =
+                    bundle.getEntities(StoredPasswordEntity.class).values()
+                    .stream()
+                    .filter(e -> e.getName().equals(storedPasswordName))
+                    .findFirst()
+                    .orElse(null);
+            if (storedPassword == null) {
+                throw new LinkerException("Could not find Stored Password for JDBC Connection: " + entity.getName() + ". Password Name: " + storedPasswordName);
+            }
+
+            // replace the expression with the password reference by name
+            entity.setPasswordRef(storedPassword.getName());
+        }
```

Just want to confirm, if nothing matches, did we want to throw any exceptions or let it fall through?